packages-ts/gauntlet-terra-contracts/src/commands/abstract/executionWrapper.ts

```diff
@@ -3,14 +3,21 @@ import { Result } from '@chainlink/gauntlet-core'
 import { TerraCommand, TransactionResponse } from '@chainlink/gauntlet-terra'
 import { AccAddress } from '@terra-money/terra.js'
 import { logger, prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { Query } from './inspectionWrapper'
 
 export type BeforeExecutionContext<Input, ContractInput> = {
   input: Input
   contractInput: ContractInput
   id: string
   contract: string
+  query: Query
+  flags: any
 }
 
+export type BeforeExecute<Input, ContractInput> = (
```

Comment:
Why is this a function, returning a function, returning a promise?

Makes more sense for the `BeforeExecute` type to be a simple function returning a promise?

Reply:
The instruction command will provide the context in what the execution will happen. Then is responsibility of the command executor to actually use the hook.
IE, multisig command don't have the context of what's happening in the underlying commands, just wants to trigger the hook